### DISCUSSION TITLE
Performance comparison with pytorch 1.x and pytorch 2.x (or mixed precision training)

### DISCUSSION THREAD

**Discussion by yeruouo** [chengchingwen/Transformers.jl#134] [General]
Pre-training and large model have become the hot spot of deep learning at present, and the system based on Transform also shines in the field of computer vision. However, large model training on large-scale data sets means huge computing costs, which has become an important factor that researchers and deep learning users have to consider.

We implemented an outstanding set of models based on Flux.jl, However, there is no documentation to explain the performance of our model pipeline in terms of computational costs, such as GPU memory required by the model in classification task, time spent on forward and reverse propagation of a mini-batch,  the time spent on a single training Epoch, FLOPs, and so on.

Should we add a performance baseline of our model in a prominent place in the document, on well-known tasks such as classification, translation, etc., to attract deep learning researchers and users to use our model library? Is there a significant performance gap between our implementation and the current popular pytoch?

If we can't provide some computational performance data, a person who is attracted will say "No" and walk away.

**Comment by mindplay-dk** [comment]
Literally the first thing I was wondering when I saw this library. 🙂👍

I've had my eye on this language for a while, and I've also been following the development of AI and LLMs pretty closely - I've always understood one of the selling points of Julia to be performance, and it's one reason I don't find Python very appealing, besides its clumsy dynamic type system, both strong points for Julia.

I know at least one fellow developer who is also put off by Python, to the point of not wanting to work with AI. If the performance of this library is comparable to the C/C++ binaries Python devs are using, that would definitely be very interesting to me. I'm sure we're not alone. 😊
